code/controller/controldelegator.cpp

```diff
@@ -0,0 +1,41 @@
+#include "controldelegator.h"
```

Comment:
Is the delegator really necessary? Seems like overcomplicating by quite a bit. You should be able to use polymorphism for hot swapping the controller types, hardcoding is rarely a good idea.